src/components/Checkout/checkout.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react';
+import { loadStripe, Stripe } from '@stripe/stripe-js';
+import { Elements } from '@stripe/react-stripe-js';
+import axios from 'axios';
+import CheckoutForm from './CheckoutForm';
+import { AppDispatch, RootState } from '../../redux/store';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCart } from '../../redux/actions/cartAction';
+import { cartData } from '../../types/cartTypes';
+
+const Checkout = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const publishableKey = import.meta.env.VITE_APP_API_STRIPE_PUBLISHABLE_KEY;
+  const [stripePromise, setStripePromise] = useState<Promise<Stripe | null> | null>(null);
+  const [clientSecret, setClientSecret] = useState('');
+  const { cart, loading } = useSelector((state: RootState) => state.cart);
```

Comment:
@maxCastro1 , @GSinseswa721 , Please use `cartId` state managed by `createOrder` reducer, This will allow unpaid orders to have correct `cartId` when buyer decides to complete the payment later.